### ISSUE TITLE
Option to choose file source.

### ISSUE THREAD

**Issue by and00r** [gnosygnu/xowa#42]
Please add minimum functionality to select path for image files. As you can expect, the size for the most basic wikipedia dump using v1 category search along with the most recent image files exceeds the typical formatted space for 128gb thumb drives. 

Selecting a custom file path for all wiki components would be ideal.
And of course with http server support.

The http server functionality works great by the way.
Using Microsoft Edge on a new Intel Atom x5 tablet (2w mobile CPU with LPDDR) has a very significant impact with performance and battery life. Edge is by far the most efficient browser for Windows. The best way I can describe Edge on Windows is something like a 'hardware browser' which uses whatever  CPU instruction set and special functionality the OS and hardware can provide.  It really is the most efficient browser available on earth, and much underrated in my personal opinion.  
But please, add custom filepath support as soon as you possibly can.
Thanks.

**Comment by gnosygnu** [comment]
> Selecting a custom file path for all wiki components would be ideal.
> And of course with http server support.

Yup. The next thing I'm going to do after Android search will be file database cleanup. I mention as much in my blog: https://gnosygnu.github.io/xowa/blog.html. Thanks.

**Comment by and00r** [comment]
Database cleanup? You mean trimming redundant or deleted images? What about custom file path? So that a user can have different parts of the wiki held across different storage mediums. I guess this is too much to ask, 

ah well.

**Comment by gnosygnu** [comment]
> What about custom file path? 

Sorry. My mistake. Was thinking about the 128 GB thumb drives.

Yes, I am planning to add custom paths for individual wikis. I need this to deal with .dmg files on Mac OS X. It's probably useful to others as well.

Keep in mind, there is a way to point XOWA to use a different directory / drive for **all** the wikis. You would run XOWA with a command-line parameter. For example:

* Unzip XOWA to X:\xowa\ where X: is your flash drive
* Copy all of English Wikipedia to D:\xowa\wiki\en.wikipedia.org where D: is a drive with more space
* Run XOWA on X: by running the following command-line
    java -jar xowa_windows_64.jar --wiki_dir D:\xowa\wiki\

But otherwise, custom file paths per wiki would be more useful. I'd say another month or so. Thanks.
